proposals/fp_interop/25-xxx-edits.txt

```diff
@@ -0,0 +1,107 @@
+      ISO/IEC 60559:2020 Floating-Point arithmetic
+      25-189r1 Requirements and specifications for interoperability with
+            C interchange and extended types
+
+
+1 Background
+============
+
+The ISO C23 specification (ISO/IEC 9899:2024) added support for
+interchange and extended floating-point types (C23 Annex H). The current
+Fortran 202Y work list WG5/N2249 includes adding interoperability with
+these types as accepted work item US05. At meeting #237 in October 2025,
+J3 accepted paper 25-178r1 as requirements and specifications for
+additional floating types.
+
+This paper provides edits for Fortran 202Y proposal US05, "Add extended
+floating-point types defined in ISO C23 to ISO_C_BINDING".
+
+2 Edits (relative to 25-007r1)
+==============================
+
+------------------------------------------------------------------------
+
+[xv] Add to "Intrinsic modules" the sentences:
+
+Named constants have been added to the ISO_C_BINDING module to allow
+interoperability with C interchange and extended floating-point formats
+defined in ISO/IEC 9899:2024 Annex H.
+
+------------------------------------------------------------------------
+
+[539:7-14] Replace 18.2.2 paragraph 4 with the following paragraphs:
+
+The values of C_FLOAT, C_DOUBLE, C_LONG_DOUBLE, C_FLOAT16, C_FLOAT32,
+C_FLOAT64, C_FLOAT128, C_FLOAT32X, C_FLOAT64X, C_FLOAT128X, C_DECIMAL32
+C_DECIMAL64, and C_DECIMAL128 LE shall each be a valid value for a real
+kind type parameter on the processor or shall be -1 if the companion
+processor's type does not have a precision equal to the precision of any
+of the Fortran processor's real kinds, -2 if the companion processor's
+type does not have a range equal to the range of any of the Fortran
+processor's real kinds, -3 if the companion processor's type has neither
+the precision nor range of any of the Fortran processor's real kinds, -4
+if there is no interoperating Fortran processor kind for other reasons,
+or -5 if the companion processor does not define the corresponding C
+type. The values of C_FLOAT_COMPLEX, C_DOUBLE_COMPLEX,
+C_LONG_DOUBLE_COMPLEX, C_FLOAT16_COMPLEX, C_FLOAT32_COMPLEX,
+C_FLOAT64_COMPLEX, C_FLOAT128_COMPLEX, C_FLOAT32X_COMPLEX,
+C_FLOAT64X_COMPLEX, and C_FLOAT128X_COMPLEX shall be the same as those
+of C_FLOAT, C_DOUBLE, C_LONG_DOUBLE, C_FLOAT16, C_FLOAT32, C_FLOAT64,
+C_FLOAT128, C_FLOAT32X, C_FLOAT64X, and C_FLOAT128X, respectively. If
+the companion processor defines the C preprocessor macros
+__STDC_IEC_60559_BFP__ and __STDC_IEC_60559_TYPES__, the values of
+C_FLOAT32 and C_FLOAT64 shall be the same as those of C_FLOAT and
+C_DOUBLE, respectively.
+
+------------------------------------------------------------------------
+
+[547:13] Replace the line
+
+      The C types mentioned in Table 18.2 are defined in ISO/IEC
+      9899:2018, 6.2.5, 7.19, and 7.20.1.
+
+with the line
+
+      The C types mentioned in Table 18.2 are defined in ISO/IEC
+      9899:2024, 6.2.5, 7.19, 7.22, and H.2.
```

Comment:
`ISO/IEC 9899:2024` is a new normative reference that needs to be added in Chapter 2.

There's also a bigger question of whether all existing references to the superseded `ISO/IEC 9899:2018` should be replaced with the new standard. Such an effort would probably be orthogonal to this work item, but it's a tangential question worth asking.